### PR TITLE
Add initial codeowners for timestamp-codeowners

### PR DESCRIPTION
**Pull request by haydentherapper**
Also cleaned up rekor-monitor to not allow force pushing.

Signed-off-by: Hayden Blauzvern <hblauzvern@google.com>

### CODE REVIEW

**Comment by haydentherapper** [comment]
cc @priyawadhwa @bobcallaway

**Comment by haydentherapper** [comment]
Added @malancas to triage for the repo

File: github-sync/github-data/users.yaml [hunk 1]
```diff
@@ -77,6 +79,10 @@ users:
         role: maintainer
       - name: tuf-root-signing-codeowners
         role: maintainer
+      - name: sigstore-oncall
```
Comment (priyawadhwa) [inline github-sync/github-data/users.yaml:82]: When I initially added TSC members to `sigstore-oncall` it overwrote the `admin` permissions we already have on the various repos, so I ended up removing it. could you remove this until we figure out why?

Comment (haydentherapper) [reply github-sync/github-data/users.yaml]: Whoops, good catch, didn't mean for this to be there! I think this showed up when I did a rebase

**Review by priyawadhwa — APPROVED** [review]

**Pull request merged**